Database_api code refactory

abitmore opened the pull request:
Changes:
* Rename full_account.hpp to api_objects.hpp
* Move API-relate objects from database_api.* into api_objects.*
* Move a util functions `price_to_string()` from database_api.* to util.*
* Create database_api_imp.hxx and move definition of `database_api_impl` class into it
* Move code around
* Wrap long lines

Putting this in 3.3 release so far, please feel free to move out if it's not appropriate.

oxarbitrage commented:
Nice stuff, in the right direction.

For continuation in new lines of functions with arguments i personally prefer to use double indentation(6 spaces in our codebase) as sometimes the namespace, type, name, etc can be so long that you will have little space to place the arguments if you place them one below the other aligned. I see you applied some of that in some cases(where you didn't had another option) with more than 6 spaces. 6 spaces will kind of work always, if more then 2 lines are needed i do something like:

https://github.com/bitshares/bitshares-core/blob/f1bedc772682a64f793271f912283976574c05e6/libraries/plugins/elasticsearch/elasticsearch_plugin.cpp#L541-L545

abitmore commented:
Rebased to latest `develop` branch.

abitmore commented:
Updated text about copyright and force-pushed to avoid having confusing copyright changes in history. 

Updated includes, built fine with GCC5.4 in Ubuntu 16.04, not sure whether there would be issue on other platforms.
